src/NServiceBus.Core/Logging/ColoredConsoleLogger.cs

```diff
@@ -1,26 +1,22 @@
 ﻿namespace NServiceBus
 {
     using System;
-    using System.Runtime.InteropServices;
     using Logging;
 
     static class ColoredConsoleLogger
     {
         static ColoredConsoleLogger()
         {
-            const int STD_OUTPUT_HANDLE = -11;
-            logToConsole = GetStdHandle(STD_OUTPUT_HANDLE) != IntPtr.Zero;
+            logToConsole = Environment.UserInteractive;
```

Comment:
compared with the GetStdHandle approach, we won't be able to detect that nsb is hosted in a WPF application as far as I understand. Seems like most approach to specifically check for an existing console are quite hacky though, so I guess we don't really have many options.